server/src/definitions/DefinitionUtils.ts

```diff
@@ -4,8 +4,8 @@
  * The whole name will be captured within a group.
  * see https://stackoverflow.com/a/51166092/3577898
  */
-const YML_ENTITY_NAME_START_REGEXP = /^((\p{Letter}|[_0-9])+)/u;
-const YML_ENTITY_NAME_END_REGEXP = /((\p{Letter}|[_0-9])+)$/u;
+const YML_ENTITY_NAME_START_REGEXP = /^((\p{Letter}|[_0-9])+(::?(\p{Letter}|[_0-9])+)*)/u;
+const YML_ENTITY_NAME_END_REGEXP = /((\p{Letter}|[_0-9])+(::?(\p{Letter}|[_0-9])+)*)$/u;
```

Comment:
LOL you could probably factorize this stuff by using a string and the `RegExp` constructor.

https://github.com/PrismJS/prism/pull/1443/files#diff-43c117e56c7e839ab9e83577778c3236R9

It might involve adding a few backslashes, but nothing prohibitive, I think.